src/main/java/com/htisolutions/poolref/controllers/ResetPasswordController.java

```diff
@@ -0,0 +1,91 @@
+package com.htisolutions.poolref.controllers;
+
+import com.htisolutions.poolref.entities.User;
+import com.htisolutions.poolref.services.ResetPasswordService;
+import com.htisolutions.poolref.viewModels.ResetPasswordNicknameViewModel;
+import com.htisolutions.poolref.viewModels.ResetPasswordPasswordViewModel;
+import com.htisolutions.poolref.viewModels.ResetPasswordQuestionViewModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.*;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+
+@Controller
+@RequestMapping("/reset-password")
+public class ResetPasswordController {
+
+    private ResetPasswordService resetPasswordService;
+    private User user;
```

Comment:
Okay, that is true. Don't use the Id in the methods, but you can get the current user out of the context anyway in each method. Not sure why we need a global